src/CSE.cpp

```diff
@@ -78,11 +88,12 @@ class GVN : public IRMutator {
     map<Expr, int, ExprCompare> shallow_numbering;
 
     Scope<int> let_substitutions;
+    bool can_lift_load;
     int number;
 
     IRCompareCache cache;
 
-    GVN() : number(0), cache(8) {}
+    GVN() : can_lift_load(true), number(0), cache(8) {}
```

Comment:
It seems the sense of the logic here should be around protecting loads, not enabling them. I.e. naming this "protect_loads_in_scope" or something would make it clearer.